kroxylicious-filters/kroxylicious-oauthbearer-validation/src/main/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerValidation.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter.oauthbearer;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.sasl.Sasl;
+
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.security.oauthbearer.OAuthBearerValidatorCallbackHandler;
+import org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslServerProvider;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.kroxylicious.proxy.filter.FilterFactory;
+import io.kroxylicious.proxy.filter.FilterFactoryContext;
+import io.kroxylicious.proxy.filter.oauthbearer.sasl.DefaultSaslManager;
+import io.kroxylicious.proxy.filter.oauthbearer.sasl.ExponentialJitterBackoffStrategy;
+import io.kroxylicious.proxy.filter.oauthbearer.sasl.RateLimitSaslManager;
+import io.kroxylicious.proxy.filter.oauthbearer.sasl.SaslManager;
+import io.kroxylicious.proxy.plugin.Plugin;
+import io.kroxylicious.proxy.plugin.PluginConfigurationException;
+import io.kroxylicious.proxy.tag.VisibleForTesting;
+
+@Plugin(configType = OauthBearerValidation.Config.class)
+public class OauthBearerValidation implements FilterFactory<OauthBearerValidation.Config, SharedContext> {
+
+    public static final String OAUTHBEARER_MECHANISM = "OAUTHBEARER";
+
+    @SuppressWarnings("unused")
+    @VisibleForTesting
+    public OauthBearerValidation() {
+        this.oauthHandler = new OAuthBearerValidatorCallbackHandler();
+    }
+
+    public OauthBearerValidation(OAuthBearerValidatorCallbackHandler oauthHandler) {
+        this.oauthHandler = oauthHandler;
+    }
+
+    private final OAuthBearerValidatorCallbackHandler oauthHandler;
+
+    static {
+        OAuthBearerSaslServerProvider.initialize();
+    }
+
+    @Override
+    public SharedContext initialize(FilterFactoryContext context, Config config) throws PluginConfigurationException {
+        try {
+            oauthHandler.configure(
+                    createConfigMap(config),
+                    OAUTHBEARER_MECHANISM,
+                    createDefaultJaasConfig());
+            var saslServer = Sasl.createSaslServer(OAUTHBEARER_MECHANISM, "kafka", null, null, oauthHandler);
```

Comment:
I didn't notice this earlier, but haven't we got a lifecycle issue here?

`SaslServer` objects are supposed to be created once per connection and then disposed when the authentication is over.   They are not designed for reuse across connections.  This is tying the creation of the creation of the `SaslServer` to the lifecycle of the FilterFactory.  I think the  `Sasl.createSaslServer` call ought to be made by the createFilter method (which is called once  when the filters are created for the connection).

Reply:
Thanks for this remark I didn't notice that. SaslServer is now created at `createFilter`